docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,153 @@
+# New Language Feature Check Integration Process [WIP]
+
+The ability to parse new language features must be implemented and
+merged before following this document. After that, we need to update
+existing checks and create new checks to support the new language features.
+This document outlines the procedures for integrating new language features into Checkstyle.
+It includes guidelines for how to discover new checks and
+existing checks that need updates to ensure that Checkstyle remains
+aligned with language enhancements.
+
+## Check Update Procedure
+
+Updating check modules to support new language features aligns Checkstyle
+with current best practices in the Java community
+and contributes to the project's ongoing evolution.
+
+### Consider Similar and Related Tokens
+
+Identify existing tokens that are similar to the new language feature tokens.
+This comparison helps to determine which existing checks may be relevant
+and require updates to support the new token. Additionally,
+we should consider the impact that a new child token may have
+on existing checks.
+
+**Examples**:
+
+- When the `RECORD_DEF` token was introduced, it was reasonable to look at all checks
+  that had `CLASS_DEF` in their acceptable tokens.
+
+- When the `TEXT_BLOCK_CONTENT` token was introduced, it was reasonable to look at all checks
+  that had `STRING_LITERAL` in their acceptable tokens.
+
+- When switch expressions were introduced, it was reasonable to look at all checks
+  that deals with expressions because `LITERAL_SWITCH` may now fall under `EXPR` token.
+  This requires analyzing the impact of this new child token on existing checks,
+  such as `InnerAssignmentCheck`
+
+### Java Enhancement Proposals (JEPS)
+
+Review Java Enhancement Proposals (JEPS) related to the new feature.
+JEPS provide detailed information about the goals
+and motivations behind new language features.
+This information can help us understand the feature better and figure out
+which checks are most likely to be impacted.
+
+**Examples**:
+
+- Unnamed variables (`_`), JEP guides us to recognize the new role of the
+  underscore for unnamed variables and avoid flagging the non-use of such variables.
+  Consequently, we should update `UnusedLocalVariableCheck` to ensure it does not
+  incorrectly violate unnamed variables.
+
+### Always Impacted Checks
```

Comment:
```suggestion
### Frequently Impacted Checks
```

I suppose that nothing is "always", as @rnveach pointed out :)

Reply:
> I suppose that nothing is "always"

That is not true, and sorry if I lead you to believe this.

We should always be using this document with new grammar changes. We just very likely won't update every check because of the new grammar.

My main complaint was when you started mixing likely (synonyms for not always) with mandatory (synonym for always) and it didn't really make sense what was trying to be conveyed. Was it really mandatory or was it just a suggestion? A great movie once said "60% of the time, it works every time.".

Reply:
@mahfouz72 I am still good to change this. Please proceed.

Reply:
Yes done it is frequently now